qiskit_experiments/framework/backend_data.py

```diff
@@ -24,6 +25,9 @@ class BackendData:
 
     def __init__(self, backend):
         """Inits the backend and verifies version"""
+        warnings.filterwarnings(
+            "ignore", message=".*qiskit.qobj.pulse_qobj.*", category=DeprecationWarning
+        )
```

Comment:
I think this is too broad of a filter. Would it work just to catch for the `_parse_additional_data` call? I think that part generates unnecessary warnings because it gets called early. Other things here only generate warnings when access pulse data for which the warning is relevant. We could filter in the tests if you want to reduce clutter in the test output.

Reply:
I added this because the terminal was getting flooded with 

> /Users/davidmckay/miniforge3/envs/qiskit_latest/lib/python3.10/site-packages/qiskit/qobj/pulse_qobj.py:273: DeprecationWarning: The class ``qiskit.qobj.pulse_qobj.PulseQobjInstruction`` is deprecated as of qiskit 1.2. It will be removed in the 2.0 release. The `Qobj` class and related functionality are part of the deprecated `BackendV1` workflow,  and no longer necessary for `BackendV2`. If a user workflow requires `Qobj` it likely relies on deprecated functionality and should be updated to use `BackendV2`.

so are you saying look at where `_parse_additional_data` is called and put a `try...catch` for the DeprecationWarning?

Reply:
Not quite `try...catch`. The warnings module has `catch_warnings` for temporarily filtering warnings, so I was thinking of changing the call below to this:

```python
        if self._v2:
            with warnings.catch_warnings(
                "ignore", message=".*qiskit.qobj.pulse_qobj.*", category=DeprecationWarning
            ):
                self._parse_additional_data()
```

so the warnings are only suppressed for that call.

The thing that is weird here is that the filter gets turned on in `__init__` the first time `BackendData` instance is created and then just stays on. I think it would be better to hide warnings on parts where we know the warning is unnecessary like the `_parse_additional_data` call that is using `PulseBackendConfiguration` whether the experiment is doing something with pulse or not. If we just want to declutter the tests I would add the filter to `test/base.py` to filter the warning from all the tests. Outside of the tests, I think it is better not to shield the user from the warning if they are using something like `drive_channel` or `drive_freqs` below which are pulse-specific features.

Reply:
solved (I think)

Reply:
Yes, my suggestion is wrong, but you got the right form. It is solved up to you being happy with the level warnings still in the test output. We could suppress more but I would do it in the test configuration unless we want to dig in and figure out where the warnings are coming from. If it is things we are doing in Experiments where we are shielding the user from Qiskit changes we can hide the warnings, but I think it might just be using Qobj classes in the tests.